### ISSUE TITLE
Black screen when running on Android

### ISSUE THREAD

**Issue by Namaneo** [chrisd1100/libmatoya#32]
As an experiment, I'm developing a RetroArch front-end based on this library. I got it working correctly on Ubuntu, with video, audio and inputs. 

Then I tried to setup an Android application using it, and I keep having a black screen on the main `Activity` (the one I used to initialize the `Matoya` class). I know the native library is running correctly because I have sound and inputs (I can see those firing up through the logs).

I'm sure the culprit does not come from `libmatoya`, and I don't think the issue could come from the Java part of the Android application (which is basically 2 lines of code, 5 if we count the `onDestroy` part). However, I couldn't find any documentation or examples explaining some kind of best-practices about Android.

Not sure what part of my code I can give you to help, honestly. Currently, I'm creating the window with `MTY_GFX_GL`, I still call `MTY_WindowMakeCurrent` even though it does not seem to have any effect on Android, and I'm drawing using those two method calls, executed in a dedicated rendering thread:
```c
MTY_WindowDrawQuad(ctx.app, 0, ctx.image, &desc);
MTY_WindowPresent(ctx.app, 0, 1);
```

Could you (someone) give some advices? In the form of an example project / the correct process flow to implement on the native part / just some tips-and-tricks to be careful of?

Fantastic library by the way, I haven't played with the C language since years and I must say it's a pleasure to go back to it through libmatoya. :slightly_smiling_face: 

Thanks in advance!

**Comment by chrisd1100** [comment]
Yeah, I would like to have an Android example, it's a little trickier than the other platforms so it's probably something small. Maybe even something in the manifest. 

As an aside, I am also using libmatoya to build a simple frontend for libretro called merton https://github.com/matoya/merton, if you're interested perhaps we can work on it together!

**Comment by Namaneo** [comment]
I would be glad to! And more, I already played few days with it: my goal was to remove ImGui as it would no fit well to Android applications (?). It worked (on Windows at least) by passing the core and the game as program parameters, but it was not very instructive from a learning perspective. I ended up studying Nanoarch instead, to actually try and use libmatoya through the migration of a program from SDL and PulseAudio.

That said, your implementation is (obviously) way more complete, I would be happy to help on its development! And in the meantime, I may have a growing list of questions regarding project architecture for this kind of cross-platform work. :blush:

Back to my issue: I looked at the Parsec Android application manifest (btw, I couldn't find if this kind of practice is unacceptable/illegal in some way for closed source programs, maybe you have an idea?), and I applied the same permissions/properties to my android application, with no luck. I also tried on my phone (instead of an emulator), and... It crashed the launcher, I had to reboot :smiley: (maybe my setup is the cause though, I had some ADB versions conflicts recently). Does it make you think of something?
